### PR TITLE
[feature](table-function)support posexplode table function

### PR DESCRIPTION
**Pull request by zhangstar333**
### What problem does this PR solve?

Issue Number: close #xxx

Related PR: #xxx

Problem Summary:
support posexplode table function
doc: https://github.com/apache/doris-website/pull/1283

### Check List (For Committer)

- Test 

    - [x] Regression test
    - [ ] Unit Test
    - [ ] Manual test (add detailed scripts or steps below)
    - [ ] No need to test or manual test. Explain why:
        - [ ] This is a refactor/code format and no logic has been changed.
        - [ ] Previous test can cover this change.
        - [ ] No colde files have been changed.
        - [ ] Other reason 

- Behavior changed:

    - [x] No.
    - [ ] Yes. 

- Does this need documentation?

    - [ ] No.
    - [x] Yes. 

- Release note

    
    
    None

### Check List (For Reviewer who merge this PR)

- [ ] Confirm the release note
- [ ] Confirm test cases
- [ ] Confirm document
- [ ] Add branch pick label

### CODE REVIEW

**Comment by zhangstar333** [comment]
run buildall

**Comment by zhangstar333** [comment]
run buildall

File: regression-test/data/nereids_p0/sql_functions/table_function/pose_explode.out [hunk 1]
```diff
@@ -0,0 +1,94 @@
+-- This file is automatically generated. You should know what you did if you want to edit this
```
Comment (cambyzju) [inline regression-test/data/nereids_p0/sql_functions/table_function/pose_explode.out:1]: duplicate `.out` file

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/generator/PosExplode.java [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+            throw new AnalysisException("only support array type for posexplode function but got "
+                    + child().getDataType());
+        }
+    }
+
+    @Override
+    public List<FunctionSignature> getSignatures() {
+        return ImmutableList.of(
+                FunctionSignature.ret(new StructType(ImmutableList.of(
+                        new StructField("col1", IntegerType.INSTANCE, true, ""),
```
Comment (cambyzju) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/generator/PosExplode.java:71]:
```suggestion
                        new StructField("pos", IntegerType.INSTANCE, true, ""),
```

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/generator/PosExplode.java [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+                        new StructField("col2", ((ArrayType) child().getDataType()).getItemType(), true, ""))))
```
Comment (cambyzju) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/generator/PosExplode.java:72]:
```suggestion
                        new StructField("col", ((ArrayType) child().getDataType()).getItemType(), true, ""))))
```

**Comment by zhangstar333** [comment]
run buildall

**Comment by zhangstar333** [comment]
run buildall

**Review by HappenLee — APPROVED** [review]
LGTM

**Review by BiteTheDDDDt — APPROVED** [review]

**Pull request merged**